http-agent/src/main/java/org/motechproject/http/agent/factory/HttpComponentsClientHttpRequestFactoryWithAuth.java

```diff
@@ -21,25 +22,31 @@
  */
 public class HttpComponentsClientHttpRequestFactoryWithAuth extends HttpComponentsClientHttpRequestFactory {
 
+    private static final String HTTP_AUTH_PREEMPTIVE = "http.auth.preemptive";
     private final String username;
     private final String password;
+    private final SettingsFacade settings;
 
-    public HttpComponentsClientHttpRequestFactoryWithAuth(String username, String password) {
+    public HttpComponentsClientHttpRequestFactoryWithAuth(String username, String password, SettingsFacade settings) {
         this.username = username;
         this.password = password;
+        this.settings = settings;
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     protected HttpContext createHttpContext(HttpMethod httpMethod, URI uri) {
-        HttpHost targetHost = new HttpHost(uri.getHost(), uri.getPort(), uri.getScheme());
-        BasicScheme basicScheme = new BasicScheme();
-        BasicAuthCache authCache = new BasicAuthCache();
-        authCache.put(targetHost, basicScheme);
         BasicHttpContext localContext = new BasicHttpContext();
-        localContext.setAttribute(ClientContext.AUTH_CACHE, authCache);
+        if (!Boolean.valueOf(settings.getProperty(HTTP_AUTH_PREEMPTIVE))) {
```

Comment:
I understood that if preemptive is true we will send POST without auth before one with auth. 
If you think your way is more intuitive I will change condition.